projects/exui-common-lib/src/lib/components/write-address-inputs/write-address-inputs.component.ts

```diff
@@ -0,0 +1,55 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { AddressModel } from '../../models';
+import { AddressOption } from '../../models/address-option.model';
+
+@Component({
+  selector: 'xuilib-write-address-inputs',
+  templateUrl: './write-address-inputs.component.html'
+})
+export class WriteAddressInputsComponent implements OnInit {
+
+  @Input()
+  public formGroup: FormGroup;
+
+  @Input()
+  public isInternational = false;
+
+  public addressFormGroup = new FormGroup({});
+
+  public countyText: string;
+  public postcodeOptional: string;
+
+  public addressOptions: AddressOption[];
+
+  public missingPostcode = false;
+
+  constructor() {
+  }
+
+  public ngOnInit(): void {
+    this.setAddressFormGroup();
+  }
+
+  private setAddressFormGroup(): void {
+    if (!this.formGroup.get('address')) {
+      this.setFormGroup();
+      return;
+    }
+    const givenAddress = this.formGroup.get('address').value;
+    givenAddress.postCode && givenAddress.postCode !== '' ? this.setFormGroup(givenAddress) : this.setFormGroup();
+  }
+
+  private setFormGroup(givenAddress?: AddressModel) {
```

Comment:
@johnbenjamin-hmcts - missing return type

Reply:
@Daniel-Lam - I have updated the code.